pkg/ledger/ledger.go

```diff
@@ -218,18 +219,16 @@ func (l *Ledger) SaveMeta(ctx context.Context, targetType string, targetID inter
 		return NewValidationError("empty target id")
 	}
 
-	lastLog, err := l.store.LastLog(ctx)
-	if err != nil {
-		return err
-	}
-
-	log := core.NewSetMetadataLog(lastLog, core.SetMetadata{
-		TargetType: strings.ToUpper(targetType),
-		TargetID:   targetID,
-		Metadata:   m,
+	err = l.store.WithTX(ctx, func(api storage.API) error {
+		switch targetType {
```

Comment:
The `if targetType != core.MetaTargetTypeTransaction && targetType != core.MetaTargetTypeAccount` above could be merged with this switch to check the type only once

Reply:
WithTx start a new sql transaction.
We should ensure all validations checks are ok before starting it as it consume a connection to the database.

Reply:
Alright!
We said together that this `if` should be replaced by a `switch` to improve readability